src/main/java/uk/gov/hmcts/ccd/domain/service/search/CaseSearchResultViewGenerator.java

```diff
@@ -61,25 +64,78 @@ public CaseSearchResultView execute(String caseTypeId,
     }
 
     private List<SearchResultViewItem> buildItems(String useCase, CaseSearchResult caseSearchResult, String caseTypeId, List<String> requestedFields) {
         CaseTypeDefinition caseTypeDefinition = getCaseTypeDefinition(caseTypeId);
         SearchResult searchResultDefinition = searchResultDefinitionService.getSearchResultDefinition(caseTypeDefinition, useCase, requestedFields);
 
         List<SearchResultViewItem> items = new ArrayList<>();
-        // Only one case type is currently supported so we can reuse the same definitions for building all items
-        caseSearchResult.getCases().forEach(caseDetails ->
-            items.add(buildSearchResultViewItem(caseDetails, caseTypeDefinition, searchResultDefinition)));
-
+        caseSearchResult.getCases().forEach(caseDetails -> {
+            filterResultsByAuthorisationAndUserRole(useCase, caseDetails, caseTypeId, requestedFields);
+            items.add(buildSearchResultViewItem(caseDetails, caseTypeDefinition, searchResultDefinition));
+        });
         return items;
     }
 
-    private List<SearchResultViewHeaderGroup> buildHeaders(String caseTypeId,
-                                                           String useCase,
-                                                           CaseSearchResult caseSearchResult,
-                                                           List<String> requestedFields) {
+    private CaseDetails filterResultsByAuthorisationAndUserRole(String useCase, CaseDetails caseDetails, String caseTypeId, List<String> requestedFields) {
+        CaseTypeDefinition caseTypeDefinition = getCaseTypeDefinition(caseTypeId);
+        caseDetails.getData().entrySet().removeIf(
+            caseField -> {
+                Optional<CaseFieldDefinition> caseFieldDefinition = caseTypeDefinition.getCaseField(caseField.getKey());
+                return !filterResultsBySearchResultsDefinition(useCase, caseTypeId, requestedFields, caseField.getKey())
+                    || !filterResultsByAuthorisationAccessOnField(caseFieldDefinition.get());
+            });
+        return caseDetails;
+    }
+
+
+    private Boolean filterResultsBySearchResultsDefinition(String useCase, String caseTypeId, List<String> requestedFields, String caseFieldId) {
+        Set<String> roles = userRepository.getUserRoles();
+        CaseTypeDefinition caseTypeDefinition = getCaseTypeDefinition(caseTypeId);
+        SearchResult searchResultDefinition = searchResultDefinitionService.getSearchResultDefinition(caseTypeDefinition, useCase, requestedFields);
+        HashMap<String, String> searchFields = getSearchResultDefinitionFieldUserRoleAndField(searchResultDefinition);
+
+        if (useCase != null) {
+            if (!searchFields.containsKey(caseFieldId)) {
+                return false;
+            }
+            String role = searchFields.get(caseFieldId);
+            if (role != null && !roles.contains(role)) {
+                return false;
+            }
+        }
```

Comment:
we could make things more reusable by adding 2 methods to `SearchResult` itself, `fieldExists(...)` and `fieldHasRole(...)` or similar

Reply:
added methods to the class as requested